source/common/upstream/upstream_impl.cc

```diff
@@ -218,6 +218,27 @@ HostVector filterHosts(const absl::node_hash_set<HostSharedPtr>& hosts,
 
 } // namespace
 
+// TODO(pianiststickman): this implementation takes a lock on the hot path and puts a copy of the
```

Comment:
I just want to call out this warning so the senior-maintainer sees it.

We have more detailed notes on how to avoid taking a lock on every request for this use-case, but @pianiststickman wants to merge this before building out the thread-local impl to establish the API even with a potential perf bottleneck and then iterate.

Note that this code can't be reached in production without adding new C++ APIs, so it will not impact existing Envoy deployments.